ppsci/arch/phylstm.py

```diff
@@ -0,0 +1,166 @@
+# Copyright (c) 2023 PaddlePaddle Authors. All Rights Reserved.
+
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+
+#     http://www.apache.org/licenses/LICENSE-2.0
+
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+import paddle
+import paddle.nn as nn
+
+from ppsci.arch import base
+
+
+class DeepPhyLSTM(base.Arch):
+    """DeepPhyLSTM init function.
+
+    Args:
+        input_size (int): The input size.
+        output_size (int): The output size.
+        hidden_size (int, optional): The hidden size. Defaults to 100.
+        model_type (int, optional): The model type, value is 2 or 3. Defaults to 2.
+
+    Examples:
+        >>> import ppsci
+        >>> model = ppsci.arch.DeepPhyLSTM(1, 1, 100)
+    """
+
+    def __init__(self, input_size, output_size, hidden_size=100, model_type=2):
+        super().__init__()
+        self.input_size = input_size
+        self.output_size = output_size
+        self.hidden_size = hidden_size
+        self.model_type = model_type
+
+        if self.model_type == 2:
+            self.lstm_model = nn.Sequential(
+                nn.LSTM(input_size, hidden_size),
+                nn.ReLU(),
+                nn.LSTM(hidden_size, hidden_size),
+                nn.ReLU(),
+                nn.LSTM(hidden_size, hidden_size),
+                nn.ReLU(),
+                nn.Linear(hidden_size, hidden_size),
+                nn.Linear(hidden_size, 3 * output_size),
+            )
+
+            self.lstm_model_f = nn.Sequential(
+                nn.LSTM(3 * output_size, hidden_size),
+                nn.ReLU(),
+                nn.LSTM(hidden_size, hidden_size),
+                nn.ReLU(),
+                nn.LSTM(hidden_size, hidden_size),
+                nn.ReLU(),
+                nn.Linear(hidden_size, hidden_size),
+                nn.Linear(hidden_size, output_size),
+            )
+
+        if self.model_type == 3:
```

Comment:
if-elif-else `raise ValueError(f"model_type should be 2 or 3, but got {model_type})"`

Reply:
1